worker/provisioner/provisioner_task.go

```diff
@@ -47,6 +47,7 @@ type ProvisionerTask interface {
 
 type MachineGetter interface {
 	Machine(names.MachineTag) (*apiprovisioner.Machine, error)
+	Machines([]names.MachineTag) ([]*apiprovisioner.Machine, []params.LifeResult, error)
 	MachinesWithTransientErrors() ([]*apiprovisioner.Machine, []params.StatusResult, error)
```

Comment:
Ah, well look at that. MachinesWithTransientErrors uses the pattern of a separate error results return param which I had a problem with in another comment. And even worse, it was me who did this 3 or so years ago :-(
Can we clean this up also as a drive by? Sigh, I suck.

Reply:
Yes, changed both.